sessionctx/stmtctx/stmtctx.go

```diff
@@ -577,6 +577,14 @@ func (sc *StatementContext) GetExecDetails() execdetails.ExecDetails {
 	return details
 }
 
+// GetExecScanProcessKeys gets the execution scan process keys for the statement.
+func (sc *StatementContext) GetExecScanProcessKeys() int64 {
+	if sc.mu.execDetails.ScanDetail != nil {
```

Comment:
Is `execDetails` be visited by multiple goroutine? I see it's in `mu.execDetails` ... in that case, maybe you need to take concurrency safe into consideration.

Reply:
Nice catch. 

I realy don't like the lock, so I do some refactor and removed this function.